homework/shelkovich_ilya/userValidationService/src/main/java/com/tutrit/java/quickstart/service/UserValidationService.java

```diff
@@ -0,0 +1,35 @@
+package com.tutrit.java.quickstart.service;
+
+import com.tutrit.java.quickstart.bean.User;
+import com.tutrit.java.quickstart.exceptions.UserValidationException;
+
+public class UserValidationService {
+
+    public boolean validate(User user) throws UserValidationException {
+        try {
+            checkUserNameLength(user);
+        } catch (UserValidationException nameException) {
+            return false;
+        }
+        try {
+            checkUserAge(user);
+        } catch (UserValidationException ageException) {
+            return false;
+        }
+        return true;
+    }
+
+    public void checkUserNameLength(User user) throws UserValidationException {
+        if (user.getFirstName().length() < 3 || user.getLastName().length() < 3) {
+            throw new UserValidationException("First or Last name is too short");
+        } else if (user.getFirstName().length() > 15 || user.getLastName().length() > 15) {
+            throw new UserValidationException("First or Last name is too long");
+        }
+    }
+
+    public void checkUserAge(User user) throws UserValidationException {
+        if (user.getAge() < 0 || user.getAge() > 120) {
+            throw new UserValidationException("Age is incorrect");
+        }
+    }
+}
```

Comment:
It's ok, but it could be better with use of Interfaces.  Let's see next time on lecture. Remind me if I'll forget.

Reply:
improved